File: service/src/main/java/mu/server/service/service/TodoService.java
package mu.server.service.service;

import mu.server.service.dto.todo.TodoUsernameResponse;
import mu.server.service.dto.todo.TodosResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TodoService {
    void saveByUserId(Long userId);

    Page<TodoUsernameResponse> findAllTodosByUsername(Pageable pageable, String username);

    Page<TodosResponse> findAllTodos(Pageable pageable);
}

File: service/src/main/java/mu/server/service/service/TodoService.kt
package mu.server.service.service

import com.blazebit.persistence.PagedList
import mu.server.persistence.repository.blaze.TodoView
import mu.server.service.dto.todo.TodoRequest
import mu.server.service.dto.todo.TodoUsernameResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface TodoService {

    fun save(todoRequest: List<TodoRequest>?, username: String?)

    fun saveByUserId(userId: Long?)

    fun findAllTodosByUsername(pageable: Pageable?, username: String?): Page<TodoUsernameResponse>?

    fun findAllTodos(pageable: Pageable): PagedList<TodoView>?

}